include/ddsrouter/yaml/implementations/YamlReader.ipp

```diff
@@ -0,0 +1,150 @@
+namespace yaml {
+
+template <typename T, YamlReaderVersion V>
+T YamlReader::get(const Yaml& yml, const TagType& tag)
+{
+    // ATTENTION: This try catch can be avoided, it is only used to add verbose information
+    try
+    {
+        return get<T, V>(get_value_in_tag(yml, tag));
+    }
+    catch(const std::exception& e)
+    {
+        throw ConfigurationException(
+            utils::Formatter() <<
+            "Error getting required value of type <" << TYPE_NAME(T) <<
+            "> in tag <" << tag << "> :\n " << e.what());
+    }
+
+    utils::tsnh(utils::Formatter() << "Impossible to arrive to this point.");
+    return get<T, V>(yml); // Unrecheable code
+}
+
+template <typename T, YamlReaderVersion V>
+T YamlReader::get_scalar(const Yaml& yml, const TagType& tag)
+{
+    return get_scalar<T, V>(get_value_in_tag(yml, tag));
+}
+
+template <typename T, YamlReaderVersion V>
+T YamlReader::get_scalar(const Yaml& yml)
+{
+    if (!yml.IsScalar())
+    {
+        throw ConfigurationException(
+            utils::Formatter() <<
+            "Trying to read a primitive value of type <" << TYPE_NAME(T) << "> from a non scalar yaml.");
+    }
+
+    try
+    {
+        return yml.as<T>();
+    }
+    catch(const std::exception& e)
+    {
+        throw ConfigurationException(
+            utils::Formatter() <<
+            "Incorrect format for primitive value, expected <" << TYPE_NAME(T) << ">:\n " << e.what());
+    }
+}
+
+template <typename T, YamlReaderVersion V>
+std::list<T> YamlReader::get_list(const Yaml& yml, const TagType& tag)
+{
+    return get_list<T, V>(get_value_in_tag(yml, tag));
+}
+
+template <typename T, YamlReaderVersion V>
+std::list<T> YamlReader::get_list(const Yaml& yml)
+{
+    if (!yml.IsSequence())
+    {
+        throw ConfigurationException(
+            utils::Formatter() << "Incorrect format, yaml Sequence expected.");
+    }
+
+    std::list<T> result;
+
+    try
+    {
+        for (Yaml element : yml)
+        {
+            result.push_back(get<T, V>(element));
+        }
+    }
+    catch(const std::exception& e)
+    {
+        throw ConfigurationException(
+            utils::Formatter() << "Error reading yaml sequence of type <" << TYPE_NAME(T) << "> :\n " << e.what());
+    }
+
+    return result;
+}
+
+template <typename T, YamlReaderVersion V>
+std::set<T> YamlReader::get_set(const Yaml& yml, const TagType& tag)
+{
+    std::list<T> elements_list = get_list<T, V>(yml, tag);
+    return std::set<T>(elements_list.begin(), elements_list.end());
+}
+
+template <typename T, YamlReaderVersion V>
+T YamlReader::get_enumeration(const Yaml& yml, const std::map<TagType, T>& enum_values)
+{
+    TagType value = get_scalar<TagType>(yml);
+
+    // Find value
+    auto it = enum_values.find(value);
+
+    if (it == enum_values.end())
+    {
+        throw ConfigurationException(
+            utils::Formatter() << "Value <" << value << "> is not valid in enumeration <" << TYPE_NAME(T) << ".");
```

Comment:
This message is added in `get_value_in_tag` method. At this point there is no tag to add into the error

Reply:
I feel this is important information if something wrong happens. In the case of `get_enumeration` the user can deduce what went wrong since the value is printed, but for `get_list` only type name is printed.

My suggestion: use try-catch in methods accepting tag.

Reply:
:+1: 
Added to every elemental method called with `tag`